src/main/java/org/pac4j/j2e/filter/RequiresAuthenticationFilter.java

```diff
@@ -81,52 +75,44 @@ public void init(final FilterConfig filterConfig) throws ServletException {
             final Config config = ConfigBuilder.build(configFactoryParam);
             ConfigSingleton.setConfig(config);
         }
-
         this.clientName = getStringParam(filterConfig, Pac4jConstants.CLIENT_NAME, this.clientName);
-        CommonHelper.assertNotNull(Pac4jConstants.CLIENT_NAME, this.clientName);
-
-        this.requireAnyRole = getStringParam(filterConfig, Pac4jConstants.REQUIRE_ANY_ROLE, this.requireAnyRole);
-        this.requireAllRoles = getStringParam(filterConfig, Pac4jConstants.REQUIRE_ALL_ROLES, this.requireAllRoles);
         this.authorizerName = getStringParam(filterConfig, Pac4jConstants.AUTHORIZER_NAME, this.authorizerName);
-
-        this.useSessionForDirectClient = getBooleanParam(filterConfig, Pac4jConstants.USE_SESSION_FOR_DIRECT_CLIENT, this.useSessionForDirectClient);
```

Comment:
Close this capability. Direct clients use the HTTP request to save / restore the user profile while indirect clients use the web session.
Changing that is possible, but only by overriding the `useSession` method.